### PR TITLE
DBC (AreaTableStore) | Maps, Vmaps, Mmaps Rewrite

### PR DESCRIPTION
**Pull request by sucofog**
**Changes proposed:**

-  
-  
-  

**Target branch(es):** 1.x/2.x etc.

**Issues addressed:** Closes #

**Tests performed:** (Does it build, tested in-game, etc)

**Known issues and TODO list:**

- [ ] 
- [ ] 

**NOTE** You no longer need to squash your commits, on merge we will squash it for you. (GitHub added a new feature)

### CODE REVIEW

**Comment by Yehonal** [comment]
We've to fix some warnings to complete the travis compilation, but it's working and tested

**Comment by Inifield** [comment]
I remember porting tools from TC and got a performance issue, pretty sure you'll have it too with this

**Comment by sucofog** [comment]
@Inifield No performance issue at all, you can use already the PR, it's tested and fixes all bugs related.

File: modules/acore/extractors/map_extractor/adt.cpp [hunk 1]
```diff
@@ -142,3 +146,8 @@ bool adt_MCLQ::prepareLoadedData()
 
     return true;
 }
+
+bool adt_MFBO::prepareLoadedData()
+{
+    return fcc == MFBOMagic.fcc;
+}
```
Comment (ShinDarth) [inline modules/acore/extractors/map_extractor/adt.cpp]: please leave a new line at the end of every file

File: modules/acore/extractors/map_extractor/adt.h [hunk 2]
```diff
@@ -293,4 +327,4 @@ class ADT_file : public FileLoader{
     adt_MHDR *a_grid;
 };
 
-#endif
+#endif
```
Comment (ShinDarth) [inline modules/acore/extractors/map_extractor/adt.h]: same here, I will not comment every line but please check them

**Pull request merged**